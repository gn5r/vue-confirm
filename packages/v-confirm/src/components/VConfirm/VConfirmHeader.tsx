import { defineComponent, toRef } from "vue";

import { useBackgroundColor, useTextColor } from "@/composables/color";
import { useThemeProps } from "@/composables/theme";

import { MdiIcon } from "@/components";
import { mdiClose } from "@mdi/js";

export default defineComponent({
  name: "v-confirm-header",
  props: {
    color: {
      type: String,
      default: "inherit",
    },
    textColor: {
      type: String,
      default: "inherit",
    },
    closeable: Boolean,
    ...useThemeProps(),
  },
  emits: ["click"],
  setup(props, { slots, emit }) {
    const { backgroundColorClasses, backgroundColorStyles } =
      useBackgroundColor(toRef(props, "color"));
    const { textColorClasses, textColorStyles } = useTextColor(
      toRef(props, "textColor")
    );

    return () => (
      <>
        <header
          class={["v-confirm__header", backgroundColorClasses.value]}
          style={[backgroundColorStyles.value]}
        >
          <div
            class={textColorClasses.value}
            style={[{ backgroundColor: "transparent" }, textColorStyles.value]}
          >
            {slots.default?.()}
          </div>
          {props.closeable ? (
            <>
              <div class="v-confirm__spacer" />
              <div class="v-confirm__close-icon">
                <MdiIcon
                  color={props.textColor}
                  dark={props.dark}
                  onClick={() => emit("click")}
                >
                  {mdiClose}
                </MdiIcon>
              </div>
            </>
          ) : null}
        </header>
      </>
    );
  },
});
