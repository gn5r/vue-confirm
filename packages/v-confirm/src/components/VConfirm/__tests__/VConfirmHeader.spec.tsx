import { mount } from "@vue/test-utils";
import { describe, it, expect } from "vitest";
import { VConfirmHeader } from "../VConfirmHeader";

describe("VConfirmHeader.tsx", () => {
  it("render a component", () => {
    const wrapper = mount(VConfirmHeader);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("not render a spacer in header when default props", () => {
    const wrapper = mount(VConfirmHeader);
    expect(wrapper.find("header > .v-confirm__spacer").exists()).toBe(false);
  });

  it("not render a close icon in header when default props", () => {
    const wrapper = mount(VConfirmHeader);
    expect(wrapper.find("header > .v-confirm__close-icon").exists()).toBe(
      false
    );
  });

  it("render a spacer in header when closeable is true", () => {
    const wrapper = mount(VConfirmHeader, { props: { closeable: true } });
    expect(wrapper.find("header > .v-confirm__spacer").exists()).toBe(true);
  });

  it("render a close icon in header when closeable is true", () => {
    const wrapper = mount(VConfirmHeader, { props: { closeable: true } });
    expect(wrapper.find("header > .v-confirm__close-icon").exists()).toBe(true);
  });

  it("should emitted click event when clicked to mdi-icon", async () => {
    const wrapper = mount(VConfirmHeader, { props: { closeable: true } });
    await wrapper
      .find("header > .v-confirm__close-icon > button")
      .trigger("click");
    expect(wrapper.emitted("click")).toBeTruthy();
  });

  it("render a title in header", () => {
    const wrapper = mount(VConfirmHeader, { slots: { default: "Title" } });
    expect(wrapper.find("header > div").text()).toBe("Title");
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("default background color of header is inherit", () => {
    const wrapper = mount(VConfirmHeader);
    expect(wrapper.find("header").attributes().style).toContain(
      "background-color: inherit"
    );
  });

  it("default text color of header is inherit", () => {
    const wrapper = mount(VConfirmHeader, { slots: { default: "Title" } });
    expect(wrapper.find("header > div").attributes().style).toContain(
      "color: inherit"
    );
  });

  it("render a header with class style background color", () => {
    const wrapper = mount(VConfirmHeader, {
      props: { color: "primary" },
      slots: { default: "Title" },
    });
    expect(wrapper.find("header").classes()).toContain("primary");
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("render a header with css style background color", () => {
    const wrapper = mount(VConfirmHeader, {
      props: { color: "#fff" },
      slots: { default: "Title" },
    });
    expect(wrapper.find("header").attributes().style).toContain(
      "background-color: rgb(255, 255, 255)"
    );
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("render a header with class style color", () => {
    const wrapper = mount(VConfirmHeader, {
      props: { textColor: "primary" },
      slots: { default: "Title" },
    });
    expect(wrapper.find("header > div").classes()).toContain("primary");
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("render a header with css style color", () => {
    const wrapper = mount(VConfirmHeader, {
      props: { textColor: "#fff" },
      slots: { default: "Title" },
    });
    expect(wrapper.find("header > div").attributes().style).toContain(
      "color: rgb(255, 255, 255)"
    );
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("render a dark MdiIcon", () => {
    const wrapper = mount(VConfirmHeader, {
      props: { closeable: true, dark: true },
    });
    expect(
      wrapper.find("header > .v-confirm__close-icon > .mdi-icon.dark").exists()
    ).toBe(true);
  });
});
