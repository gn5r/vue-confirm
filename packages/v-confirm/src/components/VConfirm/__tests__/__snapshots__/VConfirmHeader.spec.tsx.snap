// Vitest Snapshot v1

exports[`VConfirmHeader.tsx > render a component 1`] = `
"<header class=\\"v-confirm__header\\" style=\\"background-color: inherit;\\">
  <div class=\\"\\" style=\\"background-color: transparent; caret-color: inherit;\\">
    <!---->
  </div>
  <!---->
  <!---->
</header>"
`;

exports[`VConfirmHeader.tsx > render a header with class style background color 1`] = `
"<header class=\\"v-confirm__header primary\\">
  <div class=\\"\\" style=\\"background-color: transparent; caret-color: inherit;\\">Title</div>
  <!---->
  <!---->
</header>"
`;

exports[`VConfirmHeader.tsx > render a header with class style color 1`] = `
"<header class=\\"v-confirm__header\\" style=\\"background-color: inherit;\\">
  <div class=\\"primary\\" style=\\"background-color: transparent;\\">Title</div>
  <!---->
  <!---->
</header>"
`;

exports[`VConfirmHeader.tsx > render a header with css style background color 1`] = `
"<header class=\\"v-confirm__header\\" style=\\"background-color: rgb(255, 255, 255);\\">
  <div class=\\"\\" style=\\"background-color: transparent; caret-color: inherit;\\">Title</div>
  <!---->
  <!---->
</header>"
`;

exports[`VConfirmHeader.tsx > render a header with css style color 1`] = `
"<header class=\\"v-confirm__header\\" style=\\"background-color: inherit;\\">
  <div class=\\"\\" style=\\"background-color: transparent; color: rgb(255, 255, 255); caret-color: #fff;\\">Title</div>
  <!---->
  <!---->
</header>"
`;

exports[`VConfirmHeader.tsx > render a title in header 1`] = `
"<header class=\\"v-confirm__header\\" style=\\"background-color: inherit;\\">
  <div class=\\"\\" style=\\"background-color: transparent; caret-color: inherit;\\">Title</div>
  <!---->
  <!---->
</header>"
`;
